libs/core/src/schemas/entities.schemas.ts

```diff
@@ -428,3 +429,18 @@ export const ChainNode = z.object({
 
 // aliases
 export const Chain = Community;
+
+export const Outbox = z.object({
+  id: z.number(),
+  event_name: z.string(),
+  // TODO: should automatically include all event schemas
+  event_payload: z.union([
+    schemas.ThreadCreated,
+    schemas.CommentCreated,
+    schemas.GroupCreated,
+    schemas.CommunityCreated,
+  ]),
+  relayed: z.boolean(),
+  created_at: z.date(),
+  updated_at: z.date(),
```

Comment:
do we need updated at? the outbox should be append only and immutable

Reply:
`updated_at` is used for partitioning since `updated_at` will change when the `relayed` watermark is updated. If you want I could rename it to `relayed_at` for clarity...

Reply:
I see... I was expecting the watermark to be stored in a separate table (or rwlock/pesistent redis as another possibility)... thinking that:
1. no need to flag every event in the stream, just the last relayed id
2. ability to replay by just moving a single id, not all flags
3. not sure about how relayers gets deployed yet, but in case you have multiple relater processes competing for the next pull/push, you will probably need a rwlock/lease mechanism to serialize the relay

Reply:
The rwlock/lease mechanism is provided by the `SELECT FOR UPDATE` and `SKIP LOCKED` Postgres queries. The message relayer will:
1. Open a txn
2. Select events (records) `FOR UPDATE SKIP LOCKED`
3. Publish to RMQ
4. Commit txn

When an event is selected it is 'locked' and other txns that define `SKIP LOCKED` will not be able to select the same event, meaning we can run multiple message relayers without an external locking mechanism.

Only flagging the last relayed event makes it much harder to implement the top-level partition over `relayed` (would need to be a custom 'partition-like' implementation).

As for point 2 I'm not sure what you mean... We can easily replay by just giving the message relayer the event id that it should start at.

Having a separate table for watermarks would also increase complexity for little gain. First off, we have no primary key on the Outbox table so there can be no foreign key that references the Outbox id from the watermark table. Additionally, since events are eventually dropped from the table we would need to add additional logic to the archival script to remove relayed ids that no longer (in-directly) reference anything.

Having the watermark on the Outbox table doesn't significantly impact performance and greatly reduces complexity.

Reply:
more questions that come to mind:
- is the relayer making sure events are queued in the same order they were received? 
- what happens when competing relayers are processing different batches, if the order is necessary? 
- considering competing relayers a deployment detail (like restarting a new instance while old is still running) - I remember discussing we only need one logical relayer

Reply:
1. We can select + published based on insert order into Outbox table (`created_at`) so yes
2. I don't see how we could ensure enqueuing order if we have competing relayers (other than making one relayer wait for another which then defeats the entire purpose of multiple relayers). Competing relayers should only be used for events that don't have a strict order requirement. That said, I doubt we will need more than a single message relayer at any given time and the Postgres locking mechanism we are using is more of a precaution/future-proof than anything. 
3. Not sure what the question is here...